src/tests/resistance/fuzzing.rs

```diff
@@ -1153,3 +2335,43 @@ fn random_non_valid_u32(rng: &mut ChaCha8Rng, value: u32) -> u32 {
         random_value + 1
     }
 }
+
+/// Picks `n` random messages from `message_pool`, encodes them and corrupts the checksum bytes.
+fn encode_messages_and_corrupt_checksum(
+    rng: &mut ChaCha8Rng,
+    n: usize,
+    message_pool: Vec<Message>,
+) -> Vec<Vec<u8>> {
+    (0..n)
+        .map(|_| {
+            let message = message_pool.choose(rng).unwrap();
+
+            let mut buffer = vec![];
+            let mut header = message.encode(&mut buffer).unwrap();
+            header.checksum = random_non_valid_u32(rng, header.checksum);
+            header.encode(&mut buffer).unwrap();
+
+            buffer
+        })
+        .collect()
+}
+
+/// Picks `n` random messages from `message_pool`, encodes them and corrupts the body-length field
+fn encode_messages_and_corrupt_body_length_field(
+    rng: &mut ChaCha8Rng,
+    n: usize,
+    message_pool: Vec<Message>,
+) -> Vec<Vec<u8>> {
+    (0..n)
+        .map(|_| {
+            let message = message_pool.choose(rng).unwrap();
+
+            let mut buffer = vec![];
+            let mut header = message.encode(&mut buffer).unwrap();
+            header.body_length = random_non_valid_u32(rng, header.checksum);
```

Comment:
```suggestion
            header.body_length = random_non_valid_u32(rng, header.body_length);
```

Reply:
Good catch - guess I need to re-run those tests :D